src/VisualStudio/Core/Def/Implementation/CodeLens/VisualStudioCodeLensReferencesService.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeLens
+{
+    [ExportWorkspaceService(typeof(ICodeLensReferencesService), layer: ServiceLayer.Host), Shared]
+    internal sealed class VisualStudioCodeLensReferencesService : ICodeLensReferencesService
+    {
+        public async Task<ReferenceCount> GetReferenceCountAsync(Solution solution, DocumentId documentId, SyntaxNode syntaxNode, int maxSearchResults,
+            CancellationToken cancellationToken)
+        {
+            var remoteService = solution.Workspace.Services.GetService<IRemoteCodeLensReferencesService>();
```

Comment:
I needed IRemoteHostDiagnosticAnalyzerExecutor because I needed to support both dev14 and dev15. is that true for code lens? if this new code lens is only for dev15, you can just define it in .Next and depends on the fact .next.dll won't ship with dev14 so that no code lens service exist in dev14 to make sure this new code lens is disabled in dev14.